src/BlockTypes/ProductCollection.php

```diff
@@ -959,4 +964,38 @@ function( $rating ) use ( $product_visibility_terms ) {
 			),
 		);
 	}
+
+	/**
+	 * Constructs a date query for product filtering based on a specified time frame.
+	 *
+	 * @param array $time_frame {
+	 *     Associative array with 'operator' (in or not-in) and 'value' (date string).
+	 *
+	 *     @type string $operator Determines the inclusion or exclusion of the date range.
+	 *     @type string $value    The date around which the range is applied.
+	 * }
+	 * @return array Date query array; empty if parameters are invalid.
+	 */
+	private function get_date_query( array $time_frame ) : array {
+		// Validate time_frame elements.
+		if ( empty( $time_frame['operator'] ) || empty( $time_frame['value'] ) ) {
+			return array();
+		}
+
+		// Determine the query operator based on the 'operator' value.
+		$query_operator = 'in' === $time_frame['operator'] ? 'after' : 'before';
+
+		// Construct and return the date query.
+		return array(
+			'date_query' => array(
+				array(
+					'column'        => 'post_date',
```

Comment:
Would it be more reliable to use `post_date_gmt` instead of `post_date` to account for DST?

Reply:
Good catch! Using post_date_gmt would indeed make our date queries DST-proof by relying on UTC. I'll update the PR to use post_date_gmt for better consistency across time zones. 🚀

Reply:
Made the changes in https://github.com/woocommerce/woocommerce-blocks/pull/11562/commits/b807d125859b115e53aaacf59d9adec87f26a5c9